src/middlewares/errorMiddleware.js

```diff
@@ -0,0 +1,19 @@
+import { ApiError } from "../exeptions/api.error.js";
+
+export const errorMiddleware = (error, req, res, next) => {
+  if (error instanceof ApiError) {
+    res.status(error.status).send({
+      message: error.message,
+      errors: error.errors,
+    });
+  }
+
+  if (error) {
+    res.statusCode = 500;
+    res.send({
+      message: 'Server error'
+    });
+  }
+
+  next();
```

Comment:
Calling `next()` after sending a response can lead to errors or unintended behavior, as the response has already been sent to the client. Consider removing the `next()` call to prevent further middleware from executing after an error response has been sent.